### PR TITLE
Make Flag an abstract BaseClass

### PR DESCRIPTION
**Pull request by cleder**

### CODE REVIEW

**Comment by FlogFr** [comment]
Hey! Great works, thanks

One tiny question: Why the WAFFLE['FLAG_CLASS'] can't have a default to the built-in waffle Flag class model? (the one you have done in the test_app.models.Flag)

**Comment by cleder** [comment]
because BaseFlag is an abstract model, giving it a default would mean some 'magic' going on in the background. `Explicit is better than implicit.`

**Comment by FlogFr** [comment]
we could provide an optional Flag model that is not abstract if the user want to use the "default / optionnal" one?

**Comment by cleder** [comment]
PRs are welcome ;-)

File: waffle/tests/test_utils.py [hunk 1]
```diff
@@ -21,3 +23,19 @@ def test_override_settings(self):
...
+    def test_get_flag_model(self):
+        flag_model = get_flag_model()
+        assert type(flag_model) == ModelBase
+
+    @override_settings(WAFFLE={})
+    def test_get_flag_model_fails_when_unset(self):
+        with self.assertRaises(AttributeError):
+            get_flag_model()
+
+    @override_settings(WAFFLE={'FLAG_CLASS': 'test_app.NoSuchModel'})
```
Comment (komar) [inline waffle/tests/test_utils.py]: You could add test for failure when app name is wrong.

**Pull request merged**